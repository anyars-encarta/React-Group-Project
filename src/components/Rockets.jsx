import Rocket from './Rocket';
import '../styles/rockets.css';

const Rockets = () => (
  <div className="rockets">
    <Rocket />
  </div>
);

export default Rockets;
