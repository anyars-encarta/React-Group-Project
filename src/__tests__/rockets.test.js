import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import rocketsReducer from '../redux/rockets/rocketsSlice';
import Rockets from '../components/Rockets';

const rocketsStore = configureStore({
  reducer: {
    rockets: rocketsReducer,
  },
});

const data = [
  {
    id: 'Apolo1',
    name: 'id1',
    description: 'Apolo Apolo, Apolo 1',
    flickr_images: ['apolo1.png'],
    reserved: false,
  },
  {
    id: 'Apolo2',
    name: 'id2',
    description: 'Apolo Apolo, Apolo 2',
    flickr_images: ['apolo2.png'],
    reserved: false,
  },
  {
    id: 'Apolo3',
    name: 'id3',
    description: 'Apolo Apolo, Apolo 3',
    flickr_images: ['apolo3.png'],
    reserved: false,
  },
];

describe('Rockets component', () => {
  it('renders a list of rockets', async () => {
    global.fetch = jest.fn(() => Promise.resolve({ json: () => Promise.resolve(data) }));
    const { asFragment } = render(
      <Provider store={rocketsStore}>
        <Rockets />
      </Provider>,
    );

    await waitFor(() => expect(screen.getByText('id2')).toBeInTheDocument());
    await waitFor(() => expect(screen.getByText('id1')).toBeInTheDocument());
    await waitFor(() => expect(screen.getByText('id3')).toBeInTheDocument());

    expect(asFragment()).toMatchSnapshot();
  });
});
