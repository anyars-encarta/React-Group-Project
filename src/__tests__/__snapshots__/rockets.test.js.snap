// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component renders a list of rockets 1`] = `
<DocumentFragment>
  <div
    class="rockets"
  >
    <div
      class="rocket"
    >
      <img
        alt="img"
        class="rocket-img"
        src="apolo1.png"
      />
      <div
        class="rocket-info"
      >
        <h2
          class="rocket-title"
        >
          id1
        </h2>
        <p
          class="rocket-description"
        >
          Apolo Apolo, Apolo 1
        </p>
        <button
          class="btn-reserve"
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
    <div
      class="rocket"
    >
      <img
        alt="img"
        class="rocket-img"
        src="apolo2.png"
      />
      <div
        class="rocket-info"
      >
        <h2
          class="rocket-title"
        >
          id2
        </h2>
        <p
          class="rocket-description"
        >
          Apolo Apolo, Apolo 2
        </p>
        <button
          class="btn-reserve"
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
    <div
      class="rocket"
    >
      <img
        alt="img"
        class="rocket-img"
        src="apolo3.png"
      />
      <div
        class="rocket-info"
      >
        <h2
          class="rocket-title"
        >
          id3
        </h2>
        <p
          class="rocket-description"
        >
          Apolo Apolo, Apolo 3
        </p>
        <button
          class="btn-reserve"
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
